app/deliver_grant_funding/routes.py

```diff
@@ -209,6 +209,8 @@ def add_user_to_grant(grant_id: UUID) -> ResponseReturnValue:
             user = next((user for user in grant.users if user.email == form.user_email.data), None)
             if user is None:
                 created_user = interfaces.user.get_or_create_user(email_address=form.user_email.data)
+                if created_user is None:
+                    abort(404)
```

Comment:
Same here, this is a symptom of the underlying problem that creates noise and confusion - let's please remove it :)